temp_err_catcher.go

```diff
@@ -23,7 +24,8 @@ type Temporary interface {
 // ErrIsTemporary returns whether an error is Temporary(),
 // iff it implements the Temporary interface.
 func ErrIsTemporary(e error) bool {
-	te, ok := e.(Temporary)
+	var te Temporary
+	ok := errors.As(e, &te)
```

Comment:
If some temporary error `err` is wrapped with `fmt.Errorf("my wrapper: %w", err)`, this new version will correctly treat this as a temporary error.